core/ts/Poll.ts

```diff
@@ -362,6 +362,28 @@ export class Poll implements IPoll {
     }
   };
 
+  /**
+   * Updates message chain hash
+   * @param messageHash hash of message with encPubKey
+   */
+  updateChainHash = (messageHash: bigint): void => {
+    this.chainHash = hash2([this.chainHash, messageHash]);
+
+    if (this.messages.length % this.batchSizes.messageBatchSize === 0) {
+      this.batchHashes.push(this.chainHash);
+      this.currentMessageBatchIndex += 1;
+    }
+  };
+
+  /**
+   * Pad zeroes to chain hash until batch is full
```

Comment:
technically we are not padding zeroes, we are padding with the final chain hash value so worth updating the comment for clarity